yt_dlp/extractor/brilliantpala.py

```diff
@@ -27,8 +27,18 @@ def _get_logged_in_username(self, url, video_id):
             r'"username"\s*:\s*"(?P<username>[^"]+)"', webpage, 'logged-in username')
 
     def _perform_login(self, username, password):
-        login_form = self._hidden_inputs(self._download_webpage(
-            self._LOGIN_API, None, 'Downloading login page'))
+        login_page, urlh = self._download_webpage_handle(
+            self._LOGIN_API, None, 'Downloading login page', expected_status=401)
+        if not (urlh.status == 401 or urlh.url.startswith(self._LOGIN_API)):
```

Comment:
```suggestion
        if urlh.status != 401 and not urlh.url.startswith(self._LOGIN_API):
```
This could reduce cognitive overhead

Reply:
alternatively we could `try/except` here, but maybe that would be less pretty